### ISSUE TITLE
Exception in Idea 2017.3 EAP (173.3188.16)

### ISSUE THREAD

**Issue by zielu** [JChrist/gitextender#8]
```
Access is allowed from event dispatch thread only.
 EventQueue.isDispatchThread()=false isDispatchThread()=false Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@5d1ff594 Current thread: Thread[ApplicationImpl pooled thread 213,4,main] 404865485 SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 74431559

java.lang.Throwable
	at com.intellij.util.ExceptionUtil.currentStackTrace(ExceptionUtil.java:65)
	at com.intellij.psi.impl.DebugUtil.currentStackTrace(DebugUtil.java:486)
	at com.intellij.diagnostic.LogEventException.<init>(LogEventException.java:32)
	at com.intellij.openapi.application.impl.ApplicationImpl.b(ApplicationImpl.java:1094)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1083)
	at com.intellij.openapi.vcs.changes.ui.ChangesTree.updateTreeModel(ChangesTree.java:250)
	at com.intellij.openapi.vcs.changes.ui.ChangesBrowserBase$MyChangesTreeList.rebuildTree(ChangesBrowserBase.java:234)
	at com.intellij.openapi.vcs.changes.ui.SimpleChangesBrowser.setChangesToDisplay(SimpleChangesBrowser.java:56)
	at com.intellij.openapi.vcs.changes.committed.CommittedChangesTreeBrowser.c(CommittedChangesTreeBrowser.java:282)
	at com.intellij.openapi.vcs.changes.committed.CommittedChangesTreeBrowser.<init>(CommittedChangesTreeBrowser.java:147)
	at com.intellij.openapi.vcs.update.UpdateInfoTree.createCenterPanel(UpdateInfoTree.java:161)
	at com.intellij.openapi.ui.PanelWithActionsAndCloseButton.init(PanelWithActionsAndCloseButton.java:76)
	at com.intellij.openapi.vcs.update.UpdateInfoTree.<init>(UpdateInfoTree.java:122)
	at gr.jchrist.gitextender.handlers.AfterSuccessfulMergeHandler.generateUpdateInfoTree(AfterSuccessfulMergeHandler.java:100)
	at gr.jchrist.gitextender.handlers.AfterSuccessfulMergeHandler.showUpdatedFiles(AfterSuccessfulMergeHandler.java:67)
	at gr.jchrist.gitextender.handlers.AfterSuccessfulMergeHandler.afterMerge(AfterSuccessfulMergeHandler.java:36)
	at gr.jchrist.gitextender.BranchUpdater.afterMerge(BranchUpdater.java:100)
	at gr.jchrist.gitextender.BranchUpdater.merge(BranchUpdater.java:75)
	at gr.jchrist.gitextender.BranchUpdater.update(BranchUpdater.java:48)
	at gr.jchrist.gitextender.RepositoryUpdater.updateRepository(RepositoryUpdater.java:114)
	at gr.jchrist.gitextender.BackgroundableRepoUpdateTask.run(BackgroundableRepoUpdateTask.java:38)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:704)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:534)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:479)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$2.run(ProgressManagerImpl.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:320)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by JChrist** [comment]
Hi!
Thanks for the report!
The build is in all sorts of trouble on the latest 2017.3 EAP, I cannot even compile it.
There seems to be an issue with the git4idea dependency (which is a very hard requirement for this plugin).
I'll need to find the time to search this thoroughly.

**Comment by zielu** [comment]
Sure, I'm in the same boat right now :)

**Comment by sutr90** [comment]
This issue is present on windows as well:

IntelliJ IDEA 2017.3 (Ultimate Edition)
Build #IU-173.3727.127, built on November 27, 2017
Windows 10 10.0

Basically same as #9 just on windows.

**Comment by JChrist** [comment]
I have released and uploaded version 0.4.1 which hopefully :) fixes the reported issue.
Please retry with the new version (it might take a while to appear) and feel free to re-open should any issue persist.
